Dockerfile

```diff
@@ -1,16 +1,11 @@
-FROM golang:1.18
+FROM golang:1.18 as builder
 
-RUN mkdir /app
 WORKDIR /app
-
-COPY go.mod ./
-COPY go.sum ./
-RUN go mod download
-
 COPY . .
+RUN go mod download
+RUN CGO_ENABLED=0 go build -o geralt
 
-RUN go build -o /geralt
-
+FROM alpine:3.15
```

Comment:
Alpine is a very risky choice and has proven to have an illusion of security, I think Debian makes more sense in this case.

Reply:
BTW default golang image is based on Debian and I think it's not an accidental choice.

Reply:
When introducing this change I was thinking more about the resulting image size than security. The final build using pure alpine (without go etc.) is only 16mb while the image resulting from go-1.18 was about 1.1gb in size. Though the debian image is fairly coveniat for running tests. Do you know whether GCP docker registry caches the layers of the base image? If this is the cost effectivness of debian and alpine images would be the same except the base image size. Do you know if the image size affects the cold startup time? In [this](https://medium.com/google-cloud/3-ways-to-optimize-cloud-run-response-times-4504ed0e6804#:~:text=For%20starters%2C%20on%20Cloud%20Run,times%2C%20and%20slower%20deployment%20times.) post the author says that it does not but I couldn't find another confirmation of that.

Reply:
I wasn't aware that alpine was unsafe and thought that it was best to use as minimal as possible image for security to decrease the number of attack vectors. Could you show me some examples? I'd like to read up on that